src/_langs/en/updates/2015-07-22-access-ble-devices-with-web-bluetooth.markdown

```diff
@@ -0,0 +1,279 @@
+[Promises tutorial](http://www.html5rocks.com/en/tutorials/es6/promises/). One
+more thing, `() => {}` are simply ECMAScript 2015 [Arrow
+functions](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+- they have a shorter syntax compared to function expressions and lexically
+bind the `this` value.
+
+### Scan Bluetooth Devices
+
+This version of the Web Bluetooth API specification intends to allow websites,
+running in the Central role, to connect to GATT Remote Servers over either a
+Basic Rate/Enhanced Data Rate (BR/EDR) or BLE connection. It also aims to
+support communication among devices that only implement Bluetooth 4.0 or 4.1.
+
+When a website requests access to nearby BLE devices using
+`navigator.bluetooth.requestDevice`, Google Chrome will prompt user with a
+device chooser where he can pick one device or simply cancel the request. At
+the time of writing though, **the device chooser hasn't been implemented yet.**
+Only the closer device that matches filters will be returned.
+
+The `navigator.bluetooth.requestDevice` function takes a mandatory Object that
+define Bluetooth GATT services filters. These filters are used to return
+only devices that advertise these services.
+
+For instance, scanning Bluetooth devices advertising [Bluetooth GATT Battery Service](https://developer.bluetooth.org/gatt/services/Pages/ServiceViewer.aspx?u=org.bluetooth.service.battery_service.xml) is that simple.
+
+{% highlight javascript %}
+navigator.bluetooth.requestDevice({ filters: [{ services: ['battery_service'] }] })
+.then(device => {...})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+If your Bluetooth GATT Service is not the list of [the officially adopted Bluetooth
+GATT
+services](https://developer.bluetooth.org/gatt/services/Pages/ServicesHome.aspx)
+though, you may want to use the handy `window.BluetoothUUID.getService` helper
+function then. Calling it with `0x1234` will return
+`"00001234-0000-1000-8000-00805f9b34fb"`.
+
+{% highlight javascript %}
+var myService = BluetoothUUID.getService(0x1234);
+navigator.bluetooth.requestDevice({ filters: [{ services: [myService] }] })
+.then(device => {...})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+### Connect to a Bluetooth Device
+
+So what do you do now that you have a `BluetoothDevice` resolved by the
+`navigator.bluetooth.requestDevice` Promise? Let's connect to the Bluetooth
+GATT Remote Server which holds the Attribute Protocol (ATT) lookup data and
+service and characteristic definitions.
+
+{% highlight javascript %}
+navigator.bluetooth.requestDevice({ filters: [{ services: ['battery_service'] }] })
+.then(device => {
+  // Human-readable name of the device.
+  console.log(device.name);
+  // Indicates whether or not the device is paired with the system.
+  console.log(device.paired);
+  // UUIDs of protocols, profiles and services advertised by the device.
```

Comment:
For now, this only lists GATT Services. We'd include the other things if/when the spec expands to include BT Classic devices.

This list is also filtered to the UUIDs the current origin has access to. Feel free to mention that or not.